lighthouse-core/test/computed/resource-summary-test.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * @license Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const ComputedResourceSummary = require('../../computed/resource-summary.js');
+const assert = require('assert');
+
+/* eslint-env jest */
+
+function mockNetworkRecords(recordInfo) {
+  const networkRecords = recordInfo.map((info, index) => ({
+    requestId: index.toString(),
+    resourceType: info.resourceType,
+    frameId: 1,
+    finished: true,
+    priority: 'HIGH',
+    initiatorRequest: null,
+    statusCode: 200,
+    transferSize: info.transferSize,
+    url: info.url,
+  }));
+
+  return networkRecords;
+}
+
+describe('Resource summary computed', () => {
+  let networkRecords;
+  let mainResource;
+  beforeEach(() => {
+    const networkRecordInfo = [
+      {url: 'http://example.com/file.html', resourceType: 'Document', transferSize: 30},
+      {url: 'http://example.com/app.js', resourceType: 'Script', transferSize: 10},
+      {url: 'http://cdn.example.com/script.js', resourceType: 'Script', transferSize: 50},
+      {url: 'http://third-party.com/file.jpg', resourceType: 'Image', transferSize: 70},
+    ];
+    networkRecords = mockNetworkRecords(networkRecordInfo);
+    mainResource = networkRecords[0];
+  });
+
+  it('includes all resource types, regardless of whether page contains them', () => {
+    const result = ComputedResourceSummary.summarize(networkRecords, mainResource.url);
```

Comment:
These will be a little more future proof to changes in the audit if we're using closer to real network records and testing through the public interface of the file. Will you switch it over to using `network-records-to-devtools-log.js`?

It shouldn't require many changes, just some additional variables (set up in `beforeEach` or wherever)

```js
// networkRecords and mainResource set up as before
const artifacts = {
  devtoolsLog: recordsToDevtoolsLog(networkRecords),
  URL: {requestedUrl: mainResource.url, finalUrl: mainResource.url},
};
const context = {computedCache: new Map()};
// ...
const result = await ComputedResourceSummary.request(artifacts, context);
```

The current `mockNetworkRecords` will need to be changed to be compatible, dropping `frameId` (unless the `1` value is important, but I don't think it is?) and dropping `initiatorRequest` (or setting it to `undefined`...`null` isn't a valid value).